### PR TITLE
[wasm] Remove support for CommonJS

### LINKED ISSUE #63337: [wasm] Drop support for CommonJS and WasmEnableES6
During modularization of `dotnet.js` we adopted CommonJS format as transition step to ES6 modules.
Build flag `WasmEnableES6` enables us to produce either `CommonJS` or `ES6` module.

**Are there future use-cases for CommonJS module format ?** 

If not we could drop the `WasmEnableES6` flag and simplify build and test landscape.
We should also drop CommonJS samples.

Note we have `disableDotnet6Compatibility: false` switch which exports the `MONO` and `BINDING` namespaces into global namespace, in same way how Net6.0 did. This would continue working with ES6 too.

### PR DESCRIPTION
**Pull request by pavelsavara**
Fixes https://github.com/dotnet/runtime/issues/63337

It will need changes in the SDK unit tests

### CODE REVIEW

**Comment by pavelsavara** [comment]
I can use `v8 --module` to get rid of `src\mono\sample\wasm\console-v8-es6\v8shim.cjs` and make `test-main.js` ES6

File: src/mono/wasm/README.md [hunk 1]
```diff
@@ -163,11 +163,11 @@ The samples in `src/mono/sample/wasm` can be build and run like this:
 
 * console Hello world sample
 
-`dotnet build /t:RunSample console-v8-cjs/Wasm.Console.V8.CJS.Sample.csproj`
+`dotnet build /t:RunSample console-v8-es6/Wasm.Console.V8.Sample.csproj`
```
Comment (radical) [inline src/mono/wasm/README.md:166]: Does `-es6` need to be removed here?

File: src/mono/wasm/build/WasmApp.Native.targets [hunk 2]
```diff
@@ -194,7 +190,7 @@
 
       <_EmccCommonFlags Include="$(_DefaultEmccFlags)" />
       <_EmccCommonFlags Include="$(EmccFlags)" />
-      <_EmccCommonFlags Include="-s EXPORT_ES6=1"                   Condition="'$(WasmEnableES6)' != 'false'" />
+      <_EmccCommonFlags Include="-s EXPORT_ES6=1" />
```
Comment (radical) [inline src/mono/wasm/build/WasmApp.Native.targets:193]: I think this flag needs to be moved to be just for compiling? I saw emcc warnings that it is not relevant or used when linking. Doesn't block this PR.

File: src/mono/wasm/wasm.proj [hunk 3]
```diff
@@ -218,7 +214,7 @@
 
       <CMakeConfigurationLinkFlags Condition="'$(Configuration)' == 'Debug'"  >$(CMakeConfigurationEmccFlags)</CMakeConfigurationLinkFlags>
       <CMakeConfigurationLinkFlags Condition="'$(Configuration)' == 'Release'">-O2</CMakeConfigurationLinkFlags>
-      <CMakeConfigurationLinkFlags Condition="'$(WasmEnableES6)' != 'false'">$(CMakeConfigurationLinkFlags) -s EXPORT_ES6=1</CMakeConfigurationLinkFlags>
+      <CMakeConfigurationLinkFlags>$(CMakeConfigurationLinkFlags) -s EXPORT_ES6=1</CMakeConfigurationLinkFlags>
```
Comment (radical) [inline src/mono/wasm/wasm.proj:217]: super nit:
```suggestion
      <CMakeConfigurationLinkFlags                                            >$(CMakeConfigurationLinkFlags) -s EXPORT_ES6=1</CMakeConfigurationLinkFlags>
```

**Review by radical — APPROVED** [review]

File: src/mono/wasm/README.md (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/README.md:166]: shuffling folders would take more effort, let's do it later.

File: src/mono/wasm/build/WasmApp.Native.targets (hunk 2 above)
Comment (pavelsavara) [reply src/mono/wasm/build/WasmApp.Native.targets:193]: I created follow-up issue for this https://github.com/dotnet/runtime/issues/72990

**Pull request merged**